.github/workflows/helm-chart-release.yml

```diff
@@ -82,3 +82,55 @@ jobs:
           publish_dir: ./to-gh-pages
           keep_files: true
           enable_jekyll: true
+  release-oci:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      packages: write
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@main
+
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.8.0
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login Helm to the registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | helm registry login ghcr.io --username "${{ github.repository_owner }}" --password-stdin
+
+      - name: Check Helm generated values are up-to-date before releasing
+        run: |
+          make check-generated-values
+
+      - name: Generate, sign and publish charts
+        env:
+          COSIGN_EXPERIMENTAL: 1
+        shell: bash
+        run: |
+          REGISTRY="ghcr.io/$GITHUB_REPOSITORY_OWNER"
+          charts=$(find charts/ -maxdepth 1 -mindepth 1 -type d)
+          for chart in $charts; do
+          	helm package $chart
+          	chart_name=$(yq '.name' $chart/Chart.yaml | sed 's/"//g')
+          	chart_verison=$(yq '.version' $chart/Chart.yaml | sed 's/"//g')
+          	package_file="$chart_name-$chart_verison.tgz"
+          	helm push $package_file "oci://$REGISTRY/$chart"
```

Comment:
Does helm push return the digest/manifest unique url once the push is done? I'm asking because it's better to invoke `cosign sign` against an immutable reference